pkg/docker/config/config.go

```diff
@@ -353,6 +494,31 @@ func modifyJSON(sys *types.SystemContext, editor func(auths *dockerConfigFile) (
 	return nil
 }
 
+// errCredNotConfigured indicates the credential service for the registry was not configured
+var errCredNotConfigured = errors.New("Credential service configuration not found")
+
+// getRegistryCredConfigs returns the credHelper configed for the registry
+// creds_store will not be used if cred_helper configured for specified registries
+func getRegistryCredConfigs(sys *types.SystemContext, registry string) ([]string, error) {
```

Comment:
This should be a function in `sysregistriesv2`, so that `sysregistriesv2` is in control of the configuration format and semantics; not private logic that this subpackage imposes on `sysregistriesv2`, and that any other consumer would have to duplicate.

Also, the documentation of the function should say what it returns / how it should be used, not the details of the configuration semantics.